apis/compute/v1beta1/computefirewallpolicyrule_types.go

```diff
@@ -0,0 +1,160 @@
+	// +optional
+	SrcFqdns []string `json:"srcFqdns,omitempty"`
+
+	/* CIDR IP address range. Maximum number of source CIDR IP ranges allowed is 256. */
+	// +optional
+	SrcIPRanges []string `json:"srcIPRanges,omitempty"`
+
+	/* The Unicode country codes whose IP addresses will be used to match against the source of traffic. Can only be specified if DIRECTION is ingress. */
+	// +optional
+	SrcRegionCodes []string `json:"srcRegionCodes,omitempty"`
+
+	/* Name of the Google Cloud Threat Intelligence list. */
+	// +optional
+	SrcThreatIntelligences []string `json:"srcThreatIntelligences,omitempty"`
+}
+
+// +kcc:proto=google.cloud.compute.v1.FirewallPolicyRule
+type ComputeFirewallPolicyRuleSpec struct {
+	/* The Action to perform when the client connection triggers the rule. Valid actions are "allow", "deny" and "goto_next". */
+	Action string `json:"action"`
+
+	/* An optional description for this resource. */
+	// +optional
+	Description *string `json:"description,omitempty"`
+
+	/* The direction in which this rule applies. Possible values: INGRESS, EGRESS */
+	Direction string `json:"direction"`
+
+	/* Denotes whether the firewall policy rule is disabled. When set to true, the firewall policy rule is not enforced and traffic behaves as if it did not exist. If this is unspecified, the firewall policy rule will be enabled. */
+	// +optional
+	Disabled *bool `json:"disabled,omitempty"`
+
+	/* Denotes whether to enable logging for a particular rule. If logging is enabled, logs will be exported to the configured export destination in Stackdriver. Logs may be exported to BigQuery or Pub/Sub. Note: you cannot enable logging on "goto_next" rules. */
+	// +optional
+	EnableLogging *bool `json:"enableLogging,omitempty"`
+
+	/* Immutable. */
+	FirewallPolicyRef *refs.ComputeFirewallPolicyRef `json:"firewallPolicyRef"`
+
+	/* A match condition that incoming traffic is evaluated against. If it evaluates to true, the corresponding 'action' is enforced. */
+	Match *FirewallpolicyruleMatch `json:"match"`
+
+	/* Immutable. An integer indicating the priority of a rule in the list. The priority must be a positive value between 0 and 2147483647. Rules are evaluated from highest to lowest priority where 0 is the highest priority and 2147483647 is the lowest prority. */
+	Priority int64 `json:"priority"`
+
+	// +optional
+	TargetResources []*refs.ComputeNetworkRef `json:"targetResources,omitempty"`
+
+	// +optional
+	TargetServiceAccounts []*refs.IAMServiceAccountRef `json:"targetServiceAccounts,omitempty"`
+}
+
+// +kcc:proto=google.cloud.compute.v1.FirewallPolicyRule
+type ComputeFirewallPolicyRuleStatus struct {
+	commonv1alpha1.CommonStatus `json:",inline"`
+	/* Type of the resource. Always `compute#firewallPolicyRule` for firewall policy rules */
+	// +optional
+	Kind *string `json:"kind,omitempty"`
+
+	/* Calculation of the complexity of a single firewall policy rule. */
+	// +optional
+	RuleTupleCount *int64 `json:"ruleTupleCount,omitempty"`
```

Comment:
Shouldn't these be in ab ObservedState struct ?

Reply:
This is a Beta resource, and we haven't have ObservedState supported for this resource. We'll keep output only fields under status.